src/System.Runtime/tests/System/GC.cs

```diff
@@ -226,4 +226,84 @@ private class TestObject
             }
         }
     }
-}
\ No newline at end of file
+
+    [Fact]
+    public static void ReRegisterForFinalize()
+    {
+        ReRegisterForFinalizeTest.Run();
+    }
+
+    private class ReRegisterForFinalizeTest
+    {
+        public static void Run()
+        {
+            TestObject.Finalized = false;
+            CreateObject();
+
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+
+            Assert.True(TestObject.Finalized);
+        }
+
+        private static void CreateObject()
+        {
+            using (var obj = new TestObject())
+            {
+                GC.SuppressFinalize(obj);
+            }
+        }
+
+        private class TestObject : IDisposable
+        {
+            public static bool Finalized { get; set; }
+
+            ~TestObject()
+            {
+                Finalized = true;
+            }
+
+            public void Dispose()
+            {
+                GC.ReRegisterForFinalize(this);
+            }
+        }
+    }
+
+    [Fact]
+    public static void GetTotalMemoryTest_NoForceCollection()
+    {
+        GC.Collect();
+
+        byte[] bytes = new byte[50000];
+        int genBefore = GC.GetGeneration(bytes);
+
+        Assert.True(GC.GetTotalMemory(false) >= bytes.Length);
+        Assert.True(GC.GetGeneration(bytes) == genBefore);
```

Comment:
The GetGeneration approach works well for the ForceCollection test.  We do run some risk in this NoForceCollection test: if a GC were to occur between the first and second calls to GetGeneration, the object would be promoted.  But as I don't think it's possible to distinguish between a collection forced by this code and one initiated for some other reason, I'm not sure there's much that can be done about it; I suggest we leave this for now, but be prepared to delete the test if it fails with any frequency.